### PR TITLE
Fix pylint errors in gn3/authentication.py

### PR DESCRIPTION
**Pull request by arunisaac**
@BonfaceKilz This pull request fixes several pylint errors in
`gn3/authentication.py`. Could you glance through them to ensure I didn't
change any code behaviour?

Also, in function `create_group`, why is the conditional `if group_name ...`
necessary? Is it that the function is meant to fail for certain inputs?

### CODE REVIEW

**Comment by BonfaceKilz** [comment]
Arun Isaac ***@***.***> anaandika:

> @BonfaceKilz This pull request fixes several pylint errors in
> `gn3/authentication.py`. Could you glance through them to ensure I didn&#39;t
> change any code behaviour?
>
Cool.

> Also, in function `create_group`, why is the conditional `if group_name ...`
> necessary? Is it that the function is meant to fail for certain inputs?
Nice observation.  Now that I think of it, you
don't really need it.  That check could happen
where the fn is called.

> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/genenetwork/genenetwork3/pull/52
>
> -- Commit Summary --
>
>   * <a href="https://github.com/genenetwork/genenetwork3/pull/52/commits/e28c2674c2caa1aebf72974d69659e65a5c451b4">Do not use dangerous default argument [].</a>
>   * <a href="https://github.com/genenetwork/genenetwork3/pull/52/commits/470eb9be79074e3a0234514cd68307a43a7f2832">When creating a group, return None explicitly.</a>
>   * <a href="https://github.com/genenetwork/genenetwork3/pull/52/commits/d93f09b793dda571aa7f568819ac21b0c8f4557b">Reformat condition on a single line.</a>
>   * <a href="https://github.com/genenetwork/genenetwork3/pull/52/commits/83ba9106fc798e18f559daafa0dc5b7d41d1a6f1">Do not shadow global symbol uuid.</a>
>
> -- File Changes --
>
>     M gn3/authentication.py (20)
>
> -- Patch Links --
>
> https://github.com/genenetwork/genenetwork3/pull/52.patch
> https://github.com/genenetwork/genenetwork3/pull/52.diff

-- 
Bonface M. K.
D4F09EB110177E03C28E2FE1F5BBAE1E0392253F (hkp://keys.gnupg.net)
Free Software Activist
Humble GNU Emacs User | Bearer of scheme-y parens
Curator: <https://upbookclub.com> | Twitter: @BonfaceKilz

File: gn3/authentication.py [hunk 1]
```diff
@@ -145,9 +144,13 @@ def get_user_info_by_key(key: str, value: str,
 
 
 def create_group(conn: Redis, group_name: Optional[str],
-                 admin_user_uids: List = [],
-                 member_user_uids: List = []) -> Optional[Dict]:
+                 admin_user_uids: List = None,
+                 member_user_uids: List = None) -> Optional[Dict]:
```
Comment (BonfaceKilz) [inline gn3/authentication.py:148]: Since we specify a "None", the correct type should be `Optional[List]`

**Review by BonfaceKilz — COMMENTED** [review]
@arunisaac Just one minor comment. Have that updated and I will merge this.

**Comment by arunisaac** [comment]
> @arunisaac Just one minor comment. Have that updated and I will merge this.
>
>> +                 admin_user_uids: List = None,
> +                 member_user_uids: List = None) -> Optional[Dict]:
>
> Since we specify a "None", the correct type should be `Optional[List]`

Thank you, Bonface! But, how about I delete that commit and you do the
condition checking at the place of invocation of create_group instead?
That seems like the better solution.

Cheers!

**Comment by BonfaceKilz** [comment]
Yep.  This is fine with me.

**Comment by arunisaac** [comment]
Done! You can merge this and make the discussed change.

Sorry about the force push! I don't know if there is a less destructive way to
do this.

**Pull request merged**